lib/ledger-node-state-machine.js

```diff
@@ -5,35 +5,71 @@
  */
 'use strict';
 
+const _ = require('lodash');
+const jsonpatch = require('fast-json-patch');
+
 /**
  * The LedgerNodeStateMachine class exposes the state machine API.
  */
 module.exports = class LedgerNodeStateMachine {
-  constructor(ledgerNode, storage, consensus) {
+  constructor({ledgerNode}) {
     this.ledgerNode = ledgerNode;
-    this.storage = storage;
-    this.consensus = consensus;
+    this.storage = ledgerNode.storage;
   }
 
   /**
-   * Gets an object from the ledger given a objectID and a set of options.
+   * Gets a record from the ledger given a recordID and a set of options.
    *
-   * objectId - the URI of the object to fetch.
-   * options - a set of options used when retrieving the object.
-   * callback(err, object) - the callback to call when finished.
+   * [maxBlockHeight] - limit the record history to the specified block height.
+   * recordId - the URI of the object to fetch.
+   * callback(err, {meta, record}) - the callback to call when finished.
    *   err - An Error if an error occurred, null otherwise.
    */
-  get(objectId, options, callback) {
-    if(typeof options === 'function') {
-      callback = options;
-      options = {};
-    }
+  get({maxBlockHeight, recordId}, callback) {
+    // the history returned from storage will be in the proper order, however
+    // that does not mean all the operations are valid
+    this.storage.operations.getRecordHistory(
+      {maxBlockHeight, recordId}, (err, recordHistory) => {
+        if(err) {
+          return callback(err);
+        }
+        callback(null, _replayHistory({recordHistory}));
+      });
 
-    // TODO: need to check to see if stateMachine is up-to-date
-    // TODO: if not, need to run StateMutators based on config -- or this
-    //   all needs to happen in a ledger node state machine worker that is
-    //   scheduled similarly to a consensus worker
+    function _replayHistory({recordHistory}) {
+      // find the first CreateWebLedgerRecord
```

Comment:
Can we add a TODO for using plugins to handle custom operations somewhere in here?

Reply:
By custom operations you mean something other than `CreateWebLedgerRecord` and `UpdateWebLedgerRecord`?  I recall you saying that we specifically did not want to add other operation types.  Because of this recollection, I don't understand the use case in order to create the TODO, also, would an issue be better?